### PR TITLE
Refactor: replace StringInSlice with slices.Contains

### PR DESCRIPTION
**Pull request by alexandear**
This PR refactors code to use [slices.Contains](https://pkg.go.dev/golang.org/x/exp/slices#Contains) instead of `StringInSlice`, [slices.Index](https://pkg.go.dev/golang.org/x/exp/slices#Index) instead of `IndexOfStringInSlice`.

Other changes:
- remove `util.StringInSlice`and `util.IndexOfStringInSlice` from `pkg/util` package;
- remove `utils.StringInSlice` from `test/utils` package;
- remove `test.StringInSlice` from `pkg/bindings/test` package.

When we upgrade to Go 1.21, `golang.org/x/exp/slices` will be replaced by [`slices`](https://pkg.go.dev/slices) from std.

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: pkg/util/utils.go [hunk 1]
```diff
@@ -161,9 +161,11 @@ func ParseRegistryCreds(creds string) (*types.DockerAuthConfig, error) {
 	}, nil
 }
 
-// StringInSlice is deprecated, use containers/common/pkg/util/StringInSlice
-func StringInSlice(s string, sl []string) bool {
-	return util.StringInSlice(s, sl)
+// StringInSlice determines if a string is in a string slice, returns bool.
+//
+// Deprecated: Use [golang.org/x/exp/slices.Contains] instead.
+func StringInSlice(sl []string, s string) bool {
```
Comment (vrothberg) [inline pkg/util/utils.go:167]: I vote for removing the two functions.  They're only used by Podman.

**Review by vrothberg — COMMENTED** [review]
LGTM, thanks!

One comment/nit.

**Comment by rhatdan** [comment]
LGTM
/approve

**Review by Luap99 — APPROVED** [review]
/lgtm

**Pull request merged**